### ISSUE TITLE
Replace `ROUTER_BRIDGE_PLANNER_PLAN_ERROR` error code

### ISSUE THREAD

**Issue by SimonSapin** [apollographql/router#3508]
When authorization directives (https://github.com/apollographql/router/pull/3397) filter a query down to nothing (an empty selection set), the Router responds with:

```json
{
  "data": null,
  "errors": [
    {
      "message": "Unauthorized query",
      "extensions": {
        "code": "ROUTER_BRIDGE_PLANNER_PLAN_ERROR"
      }
    }
  ]
}
```

The `ROUTER_BRIDGE_PLANNER_PLAN_ERROR` error code looks out of place. Users shouldn’t need to know what router-bridge is, and it’s not even relevant to what actually went wrong.

There are two places in `apollo-router/src/plugins/authorization/mod.rs` that generate a `QueryPlannerError::PlanningErrors`. This could be replaced with a dedicated enum variant for this case, which would have its own error code.

This may also be a good time to revisit this default impl that generates error codes based on the path of name of Rust types, exposing to users what I think should be implementation details:

https://github.com/apollographql/router/blob/59df51b471f92e992695591e16e8811517d672b0/apollo-router/src/graphql.rs#L178-L185

**Comment by SimonSapin** [comment]
Perhaps this case should even use the same error code as the partially-unauthorized case.

**Comment by Geal** [comment]
fixed by c36fca41
